app/code/Magento/SalesRule/Model/Validator.php

```diff
@@ -154,15 +157,16 @@ public function __construct(
      * Init process load collection of rules for specific website,
      * customer group and coupon code
      *
-     * @param int $websiteId
-     * @param int $customerGroupId
-     * @param string $couponCode
+     * @param $websiteId
+     * @param CartInterface $quote
      * @return $this
      */
-    public function init($websiteId, $customerGroupId, $couponCode)
+    public function init($websiteId, CartInterface $quote): self
```

Comment:
I'm afraid we can't do this in a patch release. no protected and especially public methods modification. Won't be able to approve SVC
At most we could add an optional parameter which is CartInterface $quote = null
Also don't change the return type, leave self out